icrawler/downloader.py

```diff
@@ -51,15 +51,24 @@ def set_file_path(self, img_task):
         The default strategy is to save images in the img_dir, with an
         increasing 6-digit number as the filename. Users can override this
         method if need to rename the image file or store it in custom paths.
+        
+        The goal is also to preserve the extension of the file.
+        The extension is computed from the url: http://some_url/image.XXX?params=a.
+        Where XXX is the extension.
 
         Args:
             img_task: The task dict got from task_queue.
 
         Output:
             Fullpath of the image.
         """
+        old_filename = img_task['img_url'].split("/")[-1]
+        extension_with_params = old_filename.split(".")
+        extension = "jpg"
+        if len(extension_with_params) > 1:
+            extension = extension_with_params[-1].split("?")[0]
         filename = os.path.join(self.img_dir,
-                                '{:0>6d}.jpg'.format(self.fetched_num))
+                                ('{:0>6d}.'+extension).format(self.fetched_num))
```

Comment:
Thanks for your work! You can use `'{:0>6d}.{}'.format(self.fetched_num, extension)` here to avoid joining the string with `+`.